src/api/external/orcidApi.ts

```diff
@@ -1,30 +1,32 @@
 import Axios from 'axios';
-import { Dispatch } from 'redux';
+import { ORCID_USER_INFO_URL } from '../../utils/constants';
 
-import { setExternalOrcid } from '../../redux/actions/orcidActions';
-import i18n from '../../translations/i18n';
-import { ORCID_USER_INFO_URL, StatusCode } from '../../utils/constants';
-import { setNotification } from '../../redux/actions/notificationActions';
-import { NotificationVariant } from '../../types/notification.types';
+interface OrcidInfo {
+  id: string;
+}
 
-export const getOrcidInfo = (orcidAccessToken: string) => {
-  return async (dispatch: Dispatch) => {
-    try {
-      const response = await Axios({
-        method: 'POST',
-        url: ORCID_USER_INFO_URL,
-        headers: {
-          Accept: 'application/json',
-          Authorization: `Bearer ${orcidAccessToken}`,
-        },
-      });
-      if (response.status === StatusCode.OK) {
-        dispatch(setExternalOrcid(response.data.sub));
-      } else {
-        dispatch(setNotification(i18n.t('feedback:error.get_orcid', NotificationVariant.Error)));
-      }
-    } catch {
-      dispatch(setNotification(i18n.t('feedback:error.get_orcid', NotificationVariant.Error)));
-    }
-  };
+export const getOrcidInfo = async (orcidAccessToken: string) => {
+  return await Axios.post<OrcidInfo>(ORCID_USER_INFO_URL, null, {
```

Comment:
tror kanskje vi bomma litt når vi lagde en egen `ApiResponse`, og heller skulle brukt generics til Axios... skal jeg lage en tech debt task for å undersøke det?

Reply:
ja, kan godt gjøre det 👍